state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1335,8 +1346,12 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 		let mut used_pools = BTreeMap::new();
 
-		for boost_tier in SORTED_BOOST_TIERS {
-			if boost_tier as u16 > max_boost_fee_bps {
+		let sorted_boost_tiers = BoostPools::<T, I>::iter_prefix(asset)
+			.map(|(tier, _)| tier)
+			.collect::<BTreeSet<_>>();
```

Comment:
Adding a debug_assert! here that they are sorted in ascending order would be helpful